runtime/compiler/trj9/control/CompilationThread.hpp

```diff
@@ -82,35 +82,33 @@ struct CompileParameters
    {
    CompileParameters(
          TR::CompilationInfoPerThreadBase *compilationInfo,
-         TR_J9VMBase * vm,
          J9VMThread * vmThread,
          TR_OptimizationPlan * optimizationPlan,
-         TR::SegmentAllocator &scratchSegmentProvider,
-         TR::Region &dispatchRegion,
-         TR_Memory &trMemory,
          const TR::CompileIlGenRequest &ilGenRequest
       ) :
       _compilationInfo(compilationInfo),
-      _vm(vm),
+      _vm(NULL),
```

Comment:
Partial construction of objects is error prone. The CompileParam object is mostly used to pass a bunch of parameters from the main `compile()` routine to `wrappedCompile()` which is static rather than a member function.
It is also used to pass the compiler object back. However, for this purpose we could use another parameter (pointer to TR::Compilation) or we could use `TR::CompilationInfoPerThreadBase *that` object because we do `     that->setCompilation(compiler);`
My suggestion would be to eliminate `_compiler` from `CompileParameters` and create the` CompilerParameters` object close to where it's needed (just before calling `wrappedCompile`). Then change `postCompilationTasks` so that it does not need `CompileParameters`